Update data model

mwbenowitz opened the pull request:
This commit brings the epub storage/accessibility report generator into line with the established data model for the SFR data pipeline. This function was originally created before the format was standardized, and this presented problems for merging data coming from s3 and the accessibility report generator. 

The new format complies with the data model and allows for asynchronous updating of records. This is important as it allows for new records to be ingested and the more time consuming process of persisting the epub files to disk and generating a report can be run as independent processes.

mwbenowitz committed:
Update ePub parser to match new data model

Update the data input/output formats to match the new
model established for the persistence layer. This not only
improves reliability, it is necessary to include additional
identifiers to use in matching/merging returned data from this
function to the persistence layer.

The major change, in addition to adding standard data fields
to the output objects, is to include filenames as item record
identifiers. These provide unique identifiers for each source
and allow for better tracking of item-level metadata.

mwbenowitz committed:
Remove deploy component of travis for now

mwbenowitz committed:
Fix failing tests

Refactoring caused several tests to fail, this resolves
those errors (and fixes one stray variable name in
index.js)

gkallenberg commented on config/dev.env:
```diff
@@ -4,7 +4,7 @@ AWS_S3_ENDPOINT=
 
 AWS_KINESIS_ENDPOINT=
-AWS_KINESIS_STREAMNAME=sfr-epub-results-development
+AWS_KINESIS_STREAMNAME=sfr-db-ingest
```
we need to have the '-development' string applied to stream names as per our standards.

gkallenberg commented on config/local.env:
```diff
@@ -4,7 +4,7 @@ AWS_S3_ENDPOINT=
 
 AWS_KINESIS_ENDPOINT=
-AWS_KINESIS_STREAMNAME=sfr-epub-results-development
+AWS_KINESIS_STREAMNAME=sfr-db-ingest
```
same as above

gkallenberg commented on src/epubParsers.js:
```diff
@@ -52,35 +52,50 @@ exports.getBuffer = (stream) => {
 
-exports.epubStore = (fileName, itemID, updated, type, response) => {
+exports.epubStore = (partName, instanceID, updated, type, response, itemData, fileName) => {
   logger.info('Storing file in S3')
```
you might want to add some identifier information to the log message here to distinguish which file is being stored.

gkallenberg commented on test/handler.test.js:
```diff
@@ -234,17 +221,19 @@ describe('Handlers [index.js]', () => {
       checkStub.resolves('dataObject')
+      explodeStub.resolves('success')
       bufferStub.resolves('bufferObject')
+      storeStub.resolves('succeess')
```
sp on 'succeess'

gkallenberg reviewed (commented):
Just a few comments, but looks good overall.

mwbenowitz committed:
Fix kinesis stream names (adding -development)

mwbenowitz committed:
Add better logging message for file being stored

mwbenowitz committed:
Respond with metadata if an existing file is found

Even in instances where the ePub file exists in an up-to-date
format, the accompanying metadata should be returned and updated
in the database if necessary

mwbenowitz committed:
Update test to match output for up-to-date epub

mwbenowitz commented:
Made the few comments that Greg made as well as added a better output block when we find an existing epub file. The metadata that is included with the file should still be returned in case that is what has been updated. The new format allows the DB manager to take that and update the relevant item record.
